flash/tabular/classification/data/data.py

```diff
@@ -58,210 +54,142 @@ def __init__(
-        return cls(**state_dict)
+        self.set_state(LabelsState(classes))
+        self.num_classes = len(classes)
 
-    def common_load_data(self, df: DataFrame, dataset: AutoDataset):
+    def common_load_data(
+        self,
+        df: DataFrame,
+        dataset: Optional[Any] = None,
+    ):
         # impute_data
         # compute train dataset stats
         dfs = _pre_transform([df], self.num_cols, self.cat_cols, self.codes, self.mean, self.std, self.target_col,
                              self.target_codes)
 
         df = dfs[0]
 
-        dataset.num_samples = len(df)
+        if dataset is not None:
+            dataset.num_samples = len(df)
+
         cat_vars = _to_cat_vars_numpy(df, self.cat_cols)
         num_vars = _to_num_vars_numpy(df, self.num_cols)
 
-        cat_vars = np.stack(cat_vars, 1) if len(cat_vars) else np.zeros((len(self), 0))
-        num_vars = np.stack(num_vars, 1) if len(num_vars) else np.zeros((len(self), 0))
+        cat_vars = np.stack(cat_vars, 1)  # if len(cat_vars) else np.zeros((len(self), 0))
+        num_vars = np.stack(num_vars, 1)  # if len(num_vars) else np.zeros((len(self), 0))
         return df, cat_vars, num_vars
 
-    def load_data(self, df: DataFrame, dataset: AutoDataset):
-        df, cat_vars, num_vars = self.common_load_data(df, dataset)
+    def load_data(self, data: DataFrame, dataset: Optional[Any] = None):
+        df, cat_vars, num_vars = self.common_load_data(data, dataset=dataset)
         target = df[self.target_col].to_numpy().astype(np.float32 if self.is_regression else np.int64)
-        return [((c, n), t) for c, n, t in zip(cat_vars, num_vars, target)]
+        return [{
+            DefaultDataKeys.INPUT: (c, n),
+            DefaultDataKeys.TARGET: t
+        } for c, n, t in zip(cat_vars, num_vars, target)]
 
-    def predict_load_data(self, sample: Union[str, DataFrame], dataset: AutoDataset):
-        df = pd.read_csv(sample) if isinstance(sample, str) else sample
-        _, cat_vars, num_vars = self.common_load_data(df, dataset)
-        return list(zip(cat_vars, num_vars))
+    def predict_load_data(self, data: DataFrame, dataset: Optional[Any] = None):
+        _, cat_vars, num_vars = self.common_load_data(data, dataset=dataset)
+        return [{DefaultDataKeys.INPUT: (c, n)} for c, n in zip(cat_vars, num_vars)]
 
-    @classmethod
-    def from_data(
-        cls,
-        train_df: DataFrame,
-        val_df: Optional[DataFrame],
-        test_df: Optional[DataFrame],
-        predict_df: Optional[DataFrame],
-        target_col: str,
-        num_cols: List[str],
-        cat_cols: List[str],
-        is_regression: bool,
-    ) -> 'TabularPreprocess':
 
-        if train_df is None:
-            raise MisconfigurationException("train_df is required to instantiate the TabularPreprocess")
+class TabularCSVDataSource(TabularDataFrameDataSource):
 
-        dfs = [train_df]
+    def load_data(self, data: str, dataset: Optional[Any] = None):
```

Comment:
not sure if we should check somehow if the file has a valid extension

Reply:
I guess that pandas will do this for us and give a reasonable error if it's not a csv